src/status_im/extensions/ethereum.cljs

```diff
@@ -317,7 +341,8 @@
                  :address   address
                  :topics    (generate-topic topics)
                  :blockhash block-hash}]]
-    (rpc-call constants/web3-get-logs params #(map parse-log %) m)))
+
+    (rpc-call constants/web3-get-logs params (fn [results] (map #(parse-log (into [] (flatten (:topics m))) %) results)) m))) ; Get event-hints from the topics input given by the user
```

Comment:
This logic (flatten) would be better extracted as a function as it it reused. Also a comment would be nice :)

Reply:
Thanks! I'm going to update it.